### ISSUE TITLE
If you hide posts with Servant or Blacklist and they happen to author all the initial posts on your dashboard, it hides them but doesn't load any more so you're stuck with nothing

### ISSUE THREAD

**Issue by JTown-** [new-xkit/XKit#863]
Add author(s) to blacklist or hide them with Servant. 

If those hidden posts are all of the posts you'd initially see on your dashboard, it hides them but doesn't load any more posts to show so you just see a blank blue screen.

**Comment by JTown-** [comment]
My workaround for now - using Blacklist, uncheck 'Don't display blocked posts at all' then 'Show anyway' on one post -- scrolling/loading new posts is reenabled.

**Comment by JTown-** [comment]
Seems to be a fix: 

add this to the end of Run() in Blacklist

XKit.tools.add_function(function() {
			Tumblr.Events.trigger("DOMEventor:updateRect");
		}, true, "");

I'm not sure if that's the most optimal way to do it (first time poking around in xkit), but it seems to work. Blacklist tries to call Tumblr.Events.trigger("DOMEventor:updateRect"); elsewhere but I'm not sure that actually runs (it doesn't wrap it in Xkit.tools.add, which seems necessary)

**Comment by Wolvan** [comment]
You are absolutely correct with that fix of using `DOMEventor:updateRect`, see https://github.com/new-xkit/XKit/blob/master/docs/Tumblr-Dashboard-API.md#updaterect. It needs to be wrapped in an `XKit.tools.add_function` and if that is not the case, it won't work. It also shouldn't go into the run function but the `do_posts` (I think that's what the post_listener function is called).

**Comment by JTown-** [comment]
Yeah, I thought I'd fixed it, but tumblr had just given me enough posts to get scrolling working again. 

Still gets stuck when it blocks too many posts before allowing enough to get through for scrolling. Tried putting it in do_posts too, no luck.

**Comment by nightpool** [comment]
I have some code somewhere that loads new posts.... Let me see if I can
find it

**Comment by dlmarquis** [comment]
@nightpool if you find it, can you link/direct/alert me to it? I've run into a couple things recently where I've wished I had that handy.
